junebug/tests/test_logging_service.py

```diff
@@ -207,6 +210,20 @@ def test_start_stop(self):
         self.assertFalse(self.service.registered())
         self.assertEqual(self.service.logfile.closed_count, 1)
 
+    @inlineCallbacks
+    def test_dir_create(self):
+        '''If log directory already exists, make sure it is not recreated'''
+        if not os.path.exists(self.service.path):
+            os.makedirs(self.service.path, 0777)
+            self.stat1 = os.stat(self.service.path)
+            yield self.service.startService()
+            self.stat2 = os.stat(self.service.path)
+            self.assertEqual(self.stat1, self.stat2)
+            yield self.service.stopService()
```

Comment:
I think that lines 218-222 should not be part of the if statement. If the path doesn't exist, we should create it, that's fine. But we should not only run our checks if the path doesn't exist. If something changes in the future that causes that path to be created before this test runs, that means that none of these checks will be executed.